### PR TITLE
feat: [CDS-48898]: Added infra to instances mapping in service detail dashboard V2

### PR DESCRIPTION
**Pull request by Sagar-Rajput7**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/InstancesDetails/InstancesDetailsDialog.tsx [hunk 1]
```diff
@@ -36,27 +36,35 @@ export default function InstancesDetailsDialog(props: InstancesDetailsDialogProp
...
-                (cluster.lastPipelineExecutionName !== null &&
-                  cluster.lastPipelineExecutionName?.toLocaleLowerCase().indexOf(searchTerm.toLocaleLowerCase()) !== -1)
-            ).length
-        ).length
+                  cluster.clusterIdentifier?.toLocaleLowerCase().includes(searchValue)) ||
+                (cluster.instanceGroupedByPipelineExecutionList?.length &&
+                  cluster.instanceGroupedByPipelineExecutionList.some(
+                    item =>
+                      item.lastPipelineExecutionName !== null &&
+                      item.lastPipelineExecutionName?.toLocaleLowerCase().includes(searchValue)
```
Comment (rameshx) [inline src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/InstancesDetails/InstancesDetailsDialog.tsx:64]: Nit: `!== null` conditions can be removed as optional chaining is used.

File: src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancesContentV2.tsx [hunk 2]
```diff
@@ -64,37 +65,55 @@ export const isClusterData = (data: InstanceGroupedByArtifact[]): boolean => {
...
+              showInfra: true
+            }
+
+            if (entity.instanceGroupedByPipelineExecutionList) {
+              entity.instanceGroupedByPipelineExecutionList.forEach((item, idx) => {
+                tableData.push({
+                  ...commonFields,
+                  showArtifact: envShow && !index && !idx,
+                  showEnv: !index && !idx,
+                  showInfra: !idx,
```
[the remark this answers is not in the document]
Comment (Sagar-Rajput7) [reply src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancesContentV2.tsx:101]: Yes

File: src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancesContentV2.tsx [hunk 3]
```diff
@@ -320,7 +342,7 @@ export const RenderInfra: Renderer<CellProps<TableRowData>> = ({
           lineClamp={1}
           color={Color.GREY_800}
         >
-          {infraName}
+          {showInfra ? infraName : ''}
```
[the remark this answers is not in the document]
Comment (Sagar-Rajput7) [reply src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancesContentV2.tsx]: Done

File: src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancesContentV2.tsx [hunk 4]
```diff
@@ -397,7 +419,7 @@ const RenderInstances: Renderer<CellProps<TableRowData>> = ({
   TOTAL_VISIBLE_INSTANCES = tableType === TableType.PREVIEW ? 4 : 7
 
   //sending undefined as we need to pass payload conditionally
-  const lastDeployedTime = lastDeployedAt ? parseInt(lastDeployedAt) : undefined
+  const lastDeployedTime = lastDeployedAt ? lastDeployedAt : undefined
```
[the remark this answers is not in the document]
Comment (Sagar-Rajput7) [reply src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancesContentV2.tsx]: Had a discussion with BE so, we don't need this anymore, so removing this.

File: src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/InstancesDetails/InstancesDetailsDialog.tsx (hunk 1 above)
Comment (Sagar-Rajput7) [reply src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/InstancesDetails/InstancesDetailsDialog.tsx]: Done, thanks for pointing this.

**Review by rameshx — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**